cherry-pick 1.1: cli/sql: avoid check_syntax on client/server version mismatch

knz opened the pull request:
This is a toned down cherry-pick of #21119, as advised by #19445.

Fixes #19445.

Release note (cli change): client-side syntax checking is now
only automatically enabled if the client and server are running
the same version of CockroachDB. This restriction will be
lifted in CockroachDB 2.0.

cc @cockroachdb/release

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/21135)
<!-- Reviewable:end -->

knz committed:
cli/sql: avoid check_syntax on client/server version mismatch

This is a toned down cherry-pick of #21119, as advised by #19445.

Release note (cli change): client-side syntax checking is now
only automatically enabled if the client and server are running
the same version of CockroachDB. This restriction will be
lifted in CockroachDB 2.0.

bdarnell commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 0 of 1 files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/21135#-:-L1isBYP2ZlT_LhvmjdC:bnfp4nl)*
<!-- Sent from Reviewable.io -->
